src/web/components/CMP.tsx

```diff
@@ -1,35 +1,43 @@
 import React, { useState, useEffect } from 'react';
-import {
-    shouldShow,
-    setErrorHandler,
-} from '@guardian/consent-management-platform';
-import { ConsentManagementPlatform } from '@guardian/consent-management-platform/lib/ConsentManagementPlatform';
-
-type Props = {
-    cmpUi: boolean; // A switch to decide if we show CMP or not
-};
+import loadable from '@loadable/component';
+
+const ConsentManagementComponent = loadable.lib(() =>
+    import(
+        '@guardian/consent-management-platform/lib/ConsentManagementPlatform'
+    ),
+);
 
-export const CMP = ({ cmpUi }: Props) => {
+export const CMP = () => {
     const [show, setShow] = useState(false);
 
     useEffect(() => {
-        if (shouldShow()) {
-            setShow(true);
-
-            // setErrorHandler takes function to be called on errors in the CMP UI
-            setErrorHandler((errMsg: string): void => {
-                const err = new Error(errMsg);
-
-                window.guardian.modules.sentry.reportError(err, 'cmp');
-            });
-        }
+        import(
+            /* webpackChunkName: "consent-management-platform" */ '@guardian/consent-management-platform'
+        ).then(({ shouldShow, setErrorHandler }) => {
+            if (shouldShow()) {
+                setShow(true);
+
+                // setErrorHandler takes function to be called on errors in the CMP UI
+                setErrorHandler((errMsg: string): void => {
+                    const err = new Error(errMsg);
+
+                    window.guardian.modules.sentry.reportError(err, 'cmp');
+                });
+            }
+        });
     }, []);
 
     const onClose = () => setShow(false);
 
-    if (!show || !cmpUi) {
+    if (!show) {
         return null;
     }
 
-    return <ConsentManagementPlatform source="dcr" onClose={onClose} />;
+    return (
+        <ConsentManagementComponent>
+            {({ ConsentManagementPlatform }) => (
+                <ConsentManagementPlatform source="dcr" onClose={onClose} />
+            )}
+        </ConsentManagementComponent>
```

Comment:
You're using the [`lib splitting`](https://loadable-components.com/docs/library-splitting/) here, can you not just use the standard [`component splitting`](https://loadable-components.com/docs/component-splitting/) in this case for the same goal?

My understanding from the docs is that library splitting is useful for things like utils, but in this case you're importing a component?

Reply:
so its a little more complex, there are 2 main reasons:

- This component originates from a lib https://github.com/guardian/dotcom-rendering/pull/1351/files#diff-90ded79e2aa2d6deb1270cc58818484aR4

- `ConsentManagementPlatform` is not a default export

When I just used a dynamic import fro CMP, I noticed the bundle size was not as small as I would have [predicted](https://github.com/guardian/dotcom-rendering/pull/1351#issuecomment-613511842). I was only able to achive the much greater reduction in [bundle size](https://github.com/guardian/dotcom-rendering/pull/1351#issuecomment-614057994) when splitting out the lib. I therefore assume that the lib does not get implicitly split out on dynamic imports. I would need to further test this assumption however.

@oliverlloyd this should answer your question https://github.com/guardian/dotcom-rendering/pull/1351/files#r409735054